runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/stubs/ChannelCache.java

```diff
@@ -85,7 +86,9 @@ static ChannelCache forTesting(
         notification -> {
           shutdownChannel(notification.getValue());
           onChannelShutdown.run();
-        });
+        },
+        // Run the removal on the calling thread for better determinism in tests.
```

Comment:
same here

Reply:
added, this doesn't change any behavior we just want the removal to run synchronously so we don't have to rely on waiting in tests